tools/azsdk-cli/docs/process-calling.md

```diff
@@ -107,3 +110,72 @@ var result = await powershellHelper.Run(cmdOptions, cancellationToken);
 - Consider parsing `OutputDetails` to handle stderr warnings vs. stdout results.
 - Never log secrets. Prefer env vars or files for passing sensitive data; avoid putting them in `Args`.
 - Keep commands cross-platform when feasible (e.g. if some commands have a `.exe` suffix for windows).
+
+## Running Language Repository Scripts
+
+Powershell scripts in language repositories should not be called directly, but instead invoked via the `RepositoryScriptService`. This service references a config in the target language repository and looks for any script overrides that match a tag. If there is a match then a tool should invoke that script. Tools can also provide a fallback implementation.
+
+See the [repo-tooling-contract](./specs/99-repo-tooling-contract.md) spec for more details on this requirement.
+
+### When to use a script override
+
+It should be a goal that script automation living in a language repository eventually be implemented in the `azsdk` tool directly.
+Currently there is a large quantity of scripts across SDK language repos that cannot all be migrated at once, and care needs to be taken to avoid tightly coupling references to those scripts from the `azsdk` tool.
+
+A repository script override should be added if:
+
+- There is active development for the script such that tying it to an `azsdk` release would greatly slow development
+- The script is brand new and/or does not have a settled contract/behavior
+- The script location may change
+- The script may need to be quickly patched in source to unblock releases, security vulnerabilities, or other issues
+- Putting the script into `azsdk` as a generic operation for all languages would create a leaky abstraction
+
+A fallback implementation should be added if:
+
+- The operation being performed only sometimes references a language repository (e.g. typespec generation)
+- A language repository has specific edge cases while most languages use generic logic for an operation
+- The override is temporary and will be removed soon. With a fallback, removing the script won't require a new `azsdk` release
+
+### Example usage
+
+```csharp
+// inject repositoryScriptService in constructor via `IRepositoryScriptService repositoryScriptService
+
+var (invoked, result) = await repositoryScriptService.TryInvoke("CodeChecks", packagePath, new()
+    {
+        { "ServiceDirectory", serviceDirectory },
+        { "SpellCheckPublicApiSurface", true }
+    }, ct);
+
+// Optionally provide a fallback implementation or no-op instead of failing
+return invoked ?
+    new CLICheckResponse(result.ExitCode, result.Output) :
+    new CLICheckResponse(1, "", "No implementation found for CodeChecks in repository.");
+```
+
+
+The corresponding repository config will look something like:
+
+```json
+[
+  {
+    "tags": [ "CodeChecks", "azsdk_package_check_code" ],
```

Comment:
It would be nice to add a section to tell language owner how/where to find the right tag `CodeCheck` and `azsdk_package_check_code` when they need to add override script in SDK repo.